src/pybind/mgr/stats/module.py

```diff
@@ -25,10 +26,16 @@ class Module(MgrModule):
     def __init__(self, *args, **kwargs):
         super(Module, self).__init__(*args, **kwargs)
         self.fs_perf_stats = FSPerfStats(self)
+        self.prev_rank0_name = None
 
     def notify(self, notify_type, notify_id):
         if notify_type == "command":
             self.fs_perf_stats.notify(notify_id)
+        elif notify_type == "fs_map":
```

Comment:
why? can't this be done in FSPerfStats::notify?

Reply:
Because FSPerfStats::notify doesn't seem to be a callback. Having the `notify_id` empty for `fs_map` notify_type, I could still move this block of code to a function say FSPerfStats::handle_fsmap_change(). Any better suggestions?

Reply:
Another issue is `self.get('fs_map')` should be called via `module` then.

Reply:
> Because FSPerfStats::notify doesn't seem to be a callback. Having the `notify_id` empty for `fs_map` notify_type, I could still move this block of code to a function say FSPerfStats::handle_fsmap_change(). Any better suggestions?

You could have FSPerfStats::notify_cmd() and FSPerfStats::notify_fsmap().